utils/wasm-builder/src/cargo_toolchain.rs

```diff
@@ -35,6 +35,9 @@ static TOOLCHAIN_CHANNELS: &[&str] = &[
 pub(crate) struct Toolchain(String);
 
 impl Toolchain {
+    /// This is a hardcoded version of nightly toolchain, tested on our CI.
+    const PINNED_NIGHTLY_TOOLCHAIN: &'static str = "nightly-2023-09-18";
```

Comment:
can you take it from `rust-toolchain.toml` ?

Reply:
I doubt it, especially if we publish wasm builder to crates.io

Reply:
and add comment in out `rust-toolchain.toml` that if changed need to change in wasm builder as well

Reply:
> what about try to find file in the root and if not found go with const? or its too unintuitive?

I was thinking about generating code in `src/` using `build.rs`. We can try this if it doesn't lead to https://github.com/rust-lang/cargo/issues/3076

Reply:
done in last 2 commit